File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/typescript/handlers.ts
import { createDog, createCat, transferAnimal, listAnimals } from './service';
import { Dog, Cat, Shelter } from './models';

export function handleCreateDog(request: any): any {
    const name = request.name;
    return createDog(name);
}

export function handleCreateCat(request: any): any {
    const name = request.name;
    return createCat(name);
}

export function handleTransfer(request: any): number {
    const shelter = new Shelter();
    const dog = new Dog(request.name);
    transferAnimal(dog, shelter);
    return shelter.count();
}

export function handleList(request: any): any[] {
    const animals = listAnimals();
    return animals;
}

export function validate(request: any): boolean {
    if (!request.name) {
        throw new Error("name required");
    }
    return true;
}

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/python/handlers.py
from service import create_dog, create_cat, transfer_animal, list_animals
from models import Dog, Cat, Shelter


def handle_create_dog(request):
    name = request.get("name")
    return create_dog(name)


def handle_create_cat(request):
    name = request.get("name")
    return create_cat(name)


def handle_transfer(request):
    shelter = Shelter()
    dog = Dog(request.get("name"))
    transfer_animal(dog, shelter)
    return shelter.count()


def handle_list(request):
    animals = list_animals()
    return animals


def validate(request):
    if not request.get("name"):
        raise ValueError("name required")
    return True
